moving_modules.md

```diff
@@ -0,0 +1,224 @@
+Moving Non-Core Types & Providers
+=================
+
+## Index
+
+* [Goals](#goals)
+* [User Stories](#user-stories)
+* [File Path Changes](#file-path-changes)
+* [Resource Type Taxonomy](#resource-type-taxonomy)
+  * [Internal](#internal)
+  * [Core](#core)
+  * [External](#external)
+  * [Dependencies](#dependencies)
+* [Packaging](#packaging)
+
+# Goals
+
+* Remove non-core types and providers from the puppet repo and move them to modules. Doing so will reduce the surface area of puppet, decrease puppet CI cycle times, while making the extracted types and providers more accessible to community members and increasing their maintainability.
+* Continue to bundle some modules with puppet-agent during packaging so that users have a batteries-included experience.
+
+# User Stories
```

Comment:
these stories are good but they are all about end user - should we include some around the maintainership of modules post-extraction? something like 

* as an experienced puppet user, when i fix a bug in a puppet-maintained provider i use, i want to contribute that fix back upstream with a minimum of friction, so that i don't have to carry the patch to future versions of the agent

* as a puppet community member with expertise in parts of the provider ecosystem, i want to manage the flow of fixes and contributions into the providers i care about, so that contributions and module releases can happen rapidly without requiring puppet core contributors to do a full agent release.

... or so?  those might be outside scope but it'd be good to see them incorporated *somewhere* if not here.

Reply:
I'm still vague on some of the packaging details, and the community part is even farther removed, but I agree having the stories helps frame the problem space better.

Reply:
Also puppet as a gem and puppet type generate